streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -3254,10 +3254,10 @@ public void shouldSupportTableTableJoinWithKStreamToKTable() {
                 "    Processor: KSTREAM-TOTABLE-0000000004 (stores: [KSTREAM-TOTABLE-STATE-STORE-0000000005])\n" +
                 "      --> KTABLE-JOINOTHER-0000000008\n" +
                 "      <-- KSTREAM-SOURCE-0000000003\n" +
-                "    Processor: KTABLE-JOINOTHER-0000000008 (stores: [KSTREAM-TOTABLE-STATE-STORE-0000000002])\n" +
+                "    Processor: KTABLE-JOINOTHER-0000000008 (stores: [KSTREAM-TOTABLE-STATE-STORE-0000000005, KSTREAM-TOTABLE-STATE-STORE-0000000002])\n" +
```

Comment:
Just checking: adding state stores to an existing processor topology is a backwards-compatible change, right? I can't think of why it wouldn't be, but if it's not then that would mean compatibility concerns for this PR.

Reply:
Yes. We only give access to an existing store (within the same sub-topology). No compatibility concern.